src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -88,4 +96,23 @@ public void testDoPost_ExistingUser() throws IOException, ServletException {
     Mockito.verify(mockRequest).setAttribute("error", "That username is already taken.");
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
   }
+
+    @Test
+    public void testDoPost_AddsActivity() throws IOException, ServletException {
+        Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+        Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+
+        UserStore mockUserStore = Mockito.mock(UserStore.class);
+        Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
+        registerServlet.setUserStore(mockUserStore);
+
+        registerServlet.doPost(mockRequest, mockResponse);
+
+        ArgumentCaptor<Activity> activityArgumentCaptor = ArgumentCaptor.forClass(Activity.class);
+
+        Mockito.verify(mockActivityStore).addActivity(activityArgumentCaptor.capture());
+        Assert.assertEquals("test username joined!", activityArgumentCaptor.getValue().getDescription());
+        Assert.assertEquals(ActivityTypeEnum.USER_ADDED, activityArgumentCaptor.getValue().getType());
+        Mockito.verify(mockResponse).sendRedirect("/login");
```

Comment:
I should remove this verify call from here, since it is not the concern of this test method